x-pack/plugins/security_solution/public/detection_engine/rule_management/components/rule_details/three_way_diff/versions_picker/versions_picker.tsx

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useCallback, useMemo } from 'react';
+import { EuiSelect } from '@elastic/eui';
+import type { EuiSelectOption } from '@elastic/eui';
+import { BASE_OPTIONS, CURRENT_OPTIONS, TARGET_OPTIONS } from './constants';
+import type { SelectedVersions } from './constants';
+import * as i18n from './translations';
+
+interface VersionsPickerProps {
+  hasBaseVersion: boolean;
+  selectedVersions: SelectedVersions;
+  onChange: (pickedVersions: SelectedVersions) => void;
+}
+
+export function VersionsPicker({
+  hasBaseVersion,
+  selectedVersions,
+  onChange,
+}: VersionsPickerProps) {
```

Comment:
Nit: can we agree on a canonical way of defining components? The previous one was defined as an `FC<...>`:

```ts
export const CustomizedPrebuiltRuleBadge: React.FC<CustomizedPrebuiltRuleBadgeProps> = ({
```

Reply:
I don't have a strong preference. One very minor downside of using `React.FC` is that it adds `children` prop to type even for childless components like this one. In React v18 `children` are removed. We are on v17 now.

`function`-style components are a slightly more readable and can be ordered in any way, so they are a bit more flexible.

In the future we could probably add an ESLint rule to enforce for canonical way if we decide what the way is. But I am fine with either approach.

Reply:
@nikitaindik I'm fine with either too. Could be a bikeshedding topic for Tech Time when we won't have an agenda :)